runtime/slice.go

```diff
@@ -156,6 +156,58 @@ func initSliceType(map[string]*Object) {
 	SliceType.slots.Repr = &unaryOpSlot{sliceRepr}
 }
 
+func sliceIndex(f *Frame, o *Object) (i int64, raised *BaseException) {
+	if index := o.typ.slots.Index; index != nil {
+		o, raised = index.Fn(f, o)
+		if raised != nil {
+			return int64(0), raised
+		}
+	}
+	if o.isInstance(IntType) {
+		return int64(toIntUnsafe(o).Value()), nil
+	}
+	if o.isInstance(LongType) {
+		return toLongUnsafe(o).Value().Int64(), nil
+	}
+	return 0, f.RaiseType(TypeErrorType, errBadSliceIndex)
```

Comment:
I see. They are very similar though. It will be difficult for people to understand which one to use. Is there a way we can structure things so that one of these functions can go away?